src/Dfe.PlanTech.Infrastructure.Contentful/Persistence/ContentfulRepository.cs

```diff
@@ -56,10 +60,21 @@ private static string CreateErrorString(ContentfulError error)
     }
 
     public async Task<IEnumerable<TEntity>> GetEntities<TEntity>(CancellationToken cancellationToken = default)
-    => await GetEntities<TEntity>(LowerCaseFirstLetter(typeof(TEntity).Name), null, cancellationToken);
+    {
+        var contentType = LowerCaseFirstLetter(typeof(TEntity).Name);
+        var key = $"{contentType}s";
+
+        return await _cache.GetOrCreateAsync(key, async () => await GetEntities<TEntity>(contentType, null, cancellationToken)) ?? [];
+    }
 
     public async Task<IEnumerable<TEntity>> GetEntities<TEntity>(IGetEntitiesOptions options, CancellationToken cancellationToken = default)
-        => await GetEntities<TEntity>(LowerCaseFirstLetter(typeof(TEntity).Name), options, cancellationToken);
+    {
+        var contentType = LowerCaseFirstLetter(typeof(TEntity).Name);
+        var jsonOptions = JsonConvert.SerializeObject(options);
+        var key = $"{contentType}_{jsonOptions}";
```

Comment:
Is there anyway we could improve the `jsonOptions` here to instead be more expected Redis key format? Like `<contentType>_<thing>_<thing2>`

Reply:
Would help get rid of keys like this as well then I think?
<img width="238" alt="image" src="https://github.com/user-attachments/assets/f5c4b35b-14bd-4bc0-816c-19fc686ca1e9" />
<img width="300" alt="image" src="https://github.com/user-attachments/assets/d3b6915c-564e-4972-940f-25deeb2910a9" />

Reply:
@jimwashbrook - Json serialisation with system.text omits properties not on the interface, the other one works but its still pretty ugly to form a key with, so I think you're on the money with records being the better of the two
I could still go down the record route, but it touches a lot of places, and I'm questioning if its worth it, I'd prefer to follow up with it, and potentially simplify the options, because only ContentQueryEquals is ever used
I made the serialization method a bit smaller - so question being, hate it or viable?

Reply:
Records was a good shout but it still doesn't output the Content Queries properly unfortunately, I'd have to do something custom to fix it, and json route is a big ugly too
I've simplied the serialise method, and I don't think its too bad now. I'm open to revisiting records instead but I think it'd maybe suit a follow on? there are some other simplifications we could do because only the content query equals is used, though leaving the functionality there is nice, but I think this approach works and puts the keys out in a useful format

Reply:
Looks much better, good job!

Only thing I'd suggest is we change the order so that `Queries` is last, so all cached items for the same parent query should be under the same nesting, and the "name" would be what's unique about them. If that makes any sense?

E.g. currently:
<img width="602" alt="image" src="https://github.com/user-attachments/assets/e7b7f0ef-0f9d-4c9a-bba3-0a54be2bafa8" />


After:
<img width="603" alt="image" src="https://github.com/user-attachments/assets/1673b79d-9946-448e-b7d5-49a0865c3251" />

Reply:
ahh good shout!

Reply:
done